app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -429,6 +421,52 @@ class CiYaml {
     _checkExceptions(exceptions);
   }
 
+  void _verifyTargetClassicRepo(pb.Target target, List<String> exceptions) {
+    // These serve mostly as documentation.
+    assert(!isFusion, 'Expected to be called in a non-monorepo');
+    assert(target.runIf.isNotEmpty, 'Expected to be called when non-empty');
+
+    // 1. Every target must depend on .ci.yaml at the root of the repo.
+    if (!target.runIf.contains('.ci.yaml')) {
+      exceptions.add('ERROR: ${target.name} is missing `.ci.yaml` in runIf');
+    }
+
+    // 2. The engine repo must additionally depend on DEPS.
```

Comment:
Can you please leave a TODO with a link to an issue reminding us to clean this up once we're ready to archive the engine repo?

Reply:
Filed as https://github.com/flutter/flutter/issues/160874